---
title: "EMOGEA"
author: "Tobias Karakach"
date: "`r Sys.Date()`"
output: html_document
---


## Tutorial


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  echo = TRUE,
  comment = "#>"
)
```
 also sabe as
 

#1 Usage

```{r echo = T, eval = F}
#-----------------------
# (1) Overview
#----------------------
# Error Modeled Gene Expression Analysis (EMOGEA), an R package for the analysis of RNA-Seq gene expression data. EMOGEA incorporates measurement uncertainties in
# the analysis of differential expression and is specifically suited for transcriptomics studies in which low-count transcripts with small fold-changes lead to significant
# biological effects. Such transcripts include signaling mRNAs and non-coding RNAs (ncRNA) which are known to exhibit low levels of expression. The package handles missing 
# values by associating disproportionately large uncertainties to those measurements, making it particularly useful for single-cell RNA-Seq measurements. It is specifically 
# suited for ordinal data as it implements a constrained alternating least-squares (ALS) approach that allows waves of expression profiles to be visualized against the ordinal 
# variable. For differential expression analysis, EMOGEA has a much higher true positivity rate (TPR) and a vanishingly small false negativity rate (FNR) compared to common 
# approaches.  
#-------------------------
# (2) Setting up the data
# ------------------------
# We start with a count matrix where each row is a gene and each column is a sample (for bulk RNAseq) or a single cell (for scRNAseq). 
# These can be obtained by mapping read sequences to a reference genome and then counting the number of reads mapped to the exons of each gene. 
# (See, for example, the Rsubread package which does both tasks.) Alternatively, pseudo-alignment methods can be used to quantify the abundance 
# of each transcript in each sample.  Alternatively, we can also accept data arising from Affymetrix arrays. In this example we will use yeast data 
# generated by  Gierli≈Ñski et al., [PMID: 26206307].

library(EMOGEA)
data(yeastExample)

#---------- (i) Get variable ------------------
#
expressionData <- yeastExample$expressionData
metaData <- yeastExample$metaData
sampleColumn <- yeastExample$sampleColumn
conditionColumn <- yeastExample$conditionColumn

#---------- (ii) Prepare data -------------------
# This function prepares input data for the EMOGEA curve resolution function that calculates the profiles of a
# gene expression matrix measured as a function of time. The input data is an expression matrix which must
# have metadata with replicate information. If there are no replicates, the total least squares implementation
# of the algorithm cannot be performed and instead, an unweighted ordinary alternating least squares  implementation is used.
# The output is a list, prepareDataOutput, that contains: the processed expression data (expressionMatrix),
# the residual matrix (residualMatrix), and the error covariance matrix (errorCovarianceMatrix). More details of the pre-processing
# performed by this function can be found in the help file for the function itself.
#

prepareDataOutput <- prepareData(
  expressionData = expressionData,
  metaData = metaData,
  sampleColumn = sampleColumn,
  conditionColumn = conditionColumn,
  applyLogTransformation = FALSE)
#
#-----------------------------------
# (2) Maximum Likelihood Projection
# ----------------------------------
# At its most basic form, Maximul Likelihood PCA can be viewed as a superset of the classical PCA that is weighted by measurement 
# errors that de-emphasize noisy measurements. It is however more sophisticated since it incorporates measurement errors of different
# structures ranging from the basic homoscedastic (iid normal) to more complex heteroscedastic noise with different correlation structures.
# mlProjection performs this MLPCA and requires the following:
# (a) expressionMatrix: the expression matrix organized as (n X m), where n is the number of genes and m is the number of samples.This input arises from the output of prepareData() as #  decribed in 1(ii) above.
# (b) errorCovarianceMatrix: The inverse of the measurement errors, whose dimensions are (*n* X *n*). You should use the errorCovarianceMatrix output of prepareData().
# (c) numberOfComponents: the number of principal components estimated to be required to reconstruct the expression matrix without loss of generality (default =15).
# (d) maxiter: The maximum number of iterations to peform during the computation of the Maximum Likelihood. Once the max iteration is reached,the algorithms stops (default=2000).
# (e) tolerance The termination tolerance for the computation of the Maximum Likelihood. Once the improvements made by Maximum Likelihood reach the
# (f) tolerance, the algorithm stops (default =1e-10).
# (g) verbose Whether to display the information about the computation or not.(default=TRUE)
#
# output: A list consisting of U (principal components), S(eigenvalues) and V (loadings).
# The estimated matrix (estimatedMatrix) is given as t(U %*% S %*% t(V)).

projectionOutput <- mlProjection(
  expressionMatrix = prepareDataOutput$expressionMatrix,
  errorCovarianceMatrix = prepareDataOutput$errorCovarianceMatrix)
#
#-----------------------------------
# (2) Self-modelling Curve Resolution 
# ----------------------------------
#
curveResOutput <- multivariateCurveResolution(
  expressionMatrix = prepareDataOutput$expressionMatrix,
  residualMatrix = prepareDataOutput$residualMatrix)

# Curve resolution (without residual matrix)
curveResOutput <- multivariateCurveResolution(expressionMatrix = prepareDataOutput$expressionMatrix)
```










